source/fundamentals/connection.txt

```diff
@@ -0,0 +1,435 @@
+       driver will wait for a connection to open before timing out.
+
+   * - **socketTimeoutMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will wait to send or receive a request before timing out.
+
+   * - **maxIdleTimeMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will allow a pooled connection to idle before closing the
+       connection.
+
+   * - **maxLifeTimeMS**
+     - integer
+     - Specifies the maximum amount of time, in milliseconds, the Java
+       driver will continue to use a pooled connection before closing the
+       connection.
+
+   * - **safe**
+     - boolean
+     - Specifies that the driver ensures that all writes are acknowledged
+       by the connected MongoDB instance. Throws an exception on any write
+       that does not receive acknowledgment from the connected MongoDB
+       instance.
+
+   * - **journal**
+     - boolean
+     - Specifies that the driver must wait for the connected MongoDB
+       instance to group commit to the journal file on disk for all writes.
+
+   * - **w**
+     - string or integer
+     - Specifies the write concern. For more information on values, see
+       the server documentation for the :manual:`w option
+       </reference/write-concern/#w-option>`. Configuring this option
+       implicitly configures the **safe** option to ``true``.
+
+   * - **wtimeoutMS**
+     - integer
+     - Specifies a time limit, in milliseconds, for the write concern. For
+       more information, see the server documentation for the
+       :manual:`wtimeoutMS option
+       </reference/connection-string/#urioption.wtimeoutMS>`. Configuring
+       this option implicitly configures the **safe** option to ``true``.
+
+   * - **readPreference**
+     - string
+     - Specifies the read preference. For more information on values, see
+       the server documentation for the :manual:`readPreference option
+       </reference/connection-string/#urioption.readPreference>`.
+
+   * - **readPreferenceTags**
+     - string
+     - Specifies the read preference tags. For more information on values, see
+       the server documentation for the :manual:`readPreferenceTags option
+       </reference/connection-string/#urioption.readPreferenceTags>`.
+
+   * - **maxStalenessSeconds**
+     - integer
+     - Specifies, in seconds, how stale a secondary can be before the
+       driver stops communicating with that secondary. Not providing
+       a parameter or explicitly setting it to ``-1`` indicates that there
+       should be no staleness check for secondaries. The minimum value is
+       either 90 seconds or the heartbeat frequency plus 10 seconds, whichever
+       is greater. For more information, see the server documentation for the
+       :manual:`maxStalenessSeconds option
+       </reference/connection-string/#urioption.maxStalenessSeconds>`.
+
+   * - **authMechanism**
```

Comment:
I don't think we say anything about how to specify username and password, do we?

Reply:
We don't, but I've added a link to the auth section so we don't have to repeat this.